wagtail/images/fields.py

```diff
@@ -114,7 +115,38 @@ def check_image_pixel_size(self, f):
             )
 
     def to_python(self, data):
-        f = super().to_python(data)
+        """
+        Check that the file-upload field data contains a valid image (GIF, JPG,
+        PNG, etc. -- whatever Willow supports). Overridden from ImageField to us
+        Willow instead of Pillow as the image library in order to enable SVG support.
+        """
+        f = FileField.to_python(self, data)
```

Comment:
My thinking is that we're not specifically interested in `FileField.to_python`, but rather whatever `to_python` methods are in the inheritance chain up to and excluding `django.forms.fields.ImageField`. Calling `super(ImageField, self).to_python` means that if (however unlikely) another class with a `to_python` method is added to the MRO between `ImageField` and `FileField` with the appropriate `super` calls, we will pick up that behaviour.

I think that using `super` like this is a clearer indication of intent - it may not be immediately clear to a developer why we're calling `FileField.to_python`. The relationship to `ImageField` is more obvious, as we can see it in the superclasses of `WagtailImageField`.

This is reasonably subjective though, as the way you have it now is technically correct. :)

Reply:
I'm specifically not calling ImageField.to_python because it uses Pillow. I'm circumventing that inheritance here on purpose. I would strongly recommend anyone wanting to extend the behavior of images.fields.WagtailImageField should extend the class, not inject a Mixin into the parent classes.

Reply:
I understand that `ImageField.to_python` is deliberately not being called - in this context, with the current MRO of `ImageField`, `super(ImageField, self).to_python` will result in `FileField.to_python` being called. The concern is not so much that a developer using the Wagtail library will alter the MRO of `WagtailImageField`, but rather that calling `FileField.to_python` explicitly creates a dependency on the MRO of a class that is outside of our control (i.e. it is part of Django)